services/github/github-total-star.service.js

```diff
@@ -0,0 +1,245 @@
+}).required()
+
+const nodesSchema = Joi.array()
+  .items(
+    Joi.object({
+      stargazers: Joi.object({
+        totalCount: nonNegativeInteger,
+      }).required(),
+    })
+  )
+  .default([])
+
+const repositoriesSchema = Joi.object({
+  pageInfo: pageInfoSchema,
+  nodes: nodesSchema,
+}).required()
+
+const schema = Joi.object({
+  data: Joi.alternatives(
+    Joi.object({
+      user: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required(),
+    Joi.object({
+      organization: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required()
+  ).required(),
+}).required()
+
+const query = gql`
+  query fetchStars(
+    $user: String!
+    $nextCursor: String
+    $affiliations: [RepositoryAffiliation]!
+  ) {
+    user(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        ownerAffiliations: $affiliations
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+
+    organization(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+
+const affiliationsAllowedValues = [
+  'OWNER',
+  `COLLABORATOR`,
+  'ORGANIZATION_MEMBER',
+]
+/**
+ * Validates affiliations should contain combination of allowed values in any order.
+ *
+ * @param {string} value affiliation current value
+ * @param {*} helpers object to construct custom error
+ *
+ * @returns {string} valiadtion error or value unchanged
+ */
+const validateAffiliations = (value, helpers) => {
+  const values = value.split(',')
+  if (values.some(e => !affiliationsAllowedValues.includes(e))) {
+    return helpers.error('any.invalid')
+  }
+  return value
+}
+
+const queryParamSchema = Joi.object({
+  affiliations: Joi.string().default('OWNER').custom(validateAffiliations),
+}).required()
+
+module.exports = class GithubTotalStarService extends GithubAuthV4Service {
+  static defaultLabel = 'stars'
+  static category = 'social'
+
+  static route = {
+    base: 'github/stars',
+    pattern: ':user',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: "GitHub User's stars",
+      namedParams: {
+        user: 'chris48s',
+      },
+      queryParams: { affiliations: 'OWNER,COLLABORATOR' },
+      staticPreview: {
+        label: this.defaultLabel,
+        message: 54,
+        style: 'social',
+      },
+      documentation: userDocumentation,
+    },
+    {
+      title: "GitHub Org's stars",
+      namedParams: {
+        user: 'badges',
+      },
```

Comment:
We can do

```js
pattern: ':org',
namedParams: {
  org: 'badges',
},
```

here on this example so that the param is styled as `:org` in the front-end instead of `:user`, just to make it slightly clearer for users :)

Reply:
Earlier I thought same, but didn't find any hack for it. It's better now.